library/Rules/KeySet.php

```diff
@@ -87,6 +87,10 @@ public function getKeys()
      */
     private function hasValidStructure($input)
     {
+        if (!is_array($input)) {
+            return false;
```

Comment:
I didn't think about it, but indeed this is a fair approach. The `Key` also only accepts arrays, so doesn't make sense to accept `ArrayAccess` or `Iterator` instances on this one.

But maybe this is something we could think about in the future, what you think?